src/main/webapp/js/projects/react/components/ProjectsSearch.js

```diff
@@ -27,38 +34,109 @@ export default class ProjectsSearch extends React.Component {
   }
 
   /**
-   * Loads the project preview data after React has initialized the component.
-   * @override
-   */
-  componentDidMount() {
-    const fetchRequest = standardizeFetchErrors(
-        fetch(makeRelativeUrlAbsolute('/project-previews')),
-        'Failed to communicate with the server, please try again later.',
-        'Encountered a server error, please try again later.');
-
-    fetchRequest.then((response) => response.json()).then((projectPreviews) => {
-      console.log('Project Previews Received:');
-      console.log(projectPreviews);
-      this.setState({
-        isFetching: false,
-        projectPreviews,
-      });
-    }).catch((error) => basicErrorHandling(error));
-  }
-
-  /**
-   * Renders the list of project previews or a loading notification if the data
-   * is still being loaded.
+   * Renders either a list of project previews or the expanded view of a single
+   * project.
    * @return {React.Component}
    * @override
    */
   render() {
-    if (this.state.isFetching) {
-      return (
-        <h1 className="text-primary text-center mt-2">Loading projects...</h1>
-      );
-    }
-
-    return <ProjectList projectPreviews={this.state.projectPreviews} />;
+    return (
+      <Router>
+        <Switch>
+          <Route path="/:projectId">
+            <ExpandedProjectFetcher />
+          </Route>
+          <Route path="/">
+            <ProjectPreviewFetcher />
+          </Route>
+        </Switch>
+      </Router>
+    );
   }
 }
+
+/**
+ * A data fetcher for project preview data that renders the list of project
+ * previews after the data is loaded.
+ * @return {React.Component} Returns the React component.
+ */
+function ProjectPreviewFetcher() {
+  const onRender = (dataFetcher) => {
```

Comment:
```suggestion
  const onRender = (props) => {
```
Should this be `props`? It was really confusing to me, I know technically it doesn't matter what we call it but it's quite a deparature from React conventions if they are props that we don't call props :)

Reply:
Yeah it's not technically props but I can see the resemblance. `dataFetcherState` definitely makes it more clear and I'll make that change! Thanks.